### DISCUSSION TITLE
Performance in JOB PHP - Tunning question

### DISCUSSION THREAD

**Discussion by thyagopacher** [community/community#43203] [Programming Help] [answered]
### Select Topic Area

Question

### Body

Hello, I would like to know how I could optimize a JOB made in php language that takes a small range of lines from mysql to run thousands of times faster, without having the danger of running the same line more than once. However, due to restricted performance, the ideal was to limit updates to the database.

I think it would be beneficial to just do the initial select, however I need to somehow record what has already been processed and avoid the same reprocessing. Any tips on where I can write optimally and quickly without affecting or barely affecting the data stream?

At each execution of the JOB I think of executing with a limit of 2, but the execution queue will easily have 10 thousand lines waiting to be processed.

**Comment by thyagopacher** [comment]
My first thought is to have the JOB run thousands of times in parallel. Please if you have any other ideas.

**Comment by tuves** [comment, marked as answer]
Thanks for posting in the GitHub Community, @thyagopacher,

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
